ras/smtstate.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+# Author: Pavithra Prakash <pavrampu@linux.vnet.ibm.com>
+
+from avocado import Test
+from avocado.utils import process, distro
+from avocado.utils.software_manager import SoftwareManager
+
+
+class smtstate_tool(Test):
+
+    '''
+    smtstate tool allows to save the current smt value and restore it.  
+    '''
+
+    def setUp(self):
+
+        sm = SoftwareManager()
+        self.detected_distro = distro.detect()
+        if not sm.check_installed("powerpc-utils") and not sm.install("powerpc-utils"):
+            self.cancel("powerpc-utils is needed for the test to be run")
+        self.distro_name = self.detected_distro.name
+        if self.distro_name == "rhel":
+            if (self.detected_distro.version < "8" or self.detected_distro.release < "4"):
+                self.cancel("smtstate tool is supported only after rhel8.4")
+        elif self.distro_name == "SuSE":
+            if (self.detected_distro.version < 15 or self.detected_distro.release < 3):
+                self.cancel("smtstate tool is supported only after sles15 sp3")
+        else:
+            self.cancel("Test case is supported only on RHEL and SLES")
+
+    def test(self):
+
+        process.system("ppc64_cpu --smt=on")
+        for i in [ "off", "on", 4, 2 ]:
+            for j in [ 2, 4, "on", "off" ]:
+                process.system("ppc64_cpu --smt=%s" % j)
+                smt_initial = process.system_output("ppc64_cpu --smt", shell=True)
+                process.system("smtstate --save")
```

Comment:
if this command failed is next steps make sense , i say we need to check and fail the test if command failed